spec/models/user_spec.rb

```diff
@@ -8,26 +8,35 @@
   describe '.from_merged_attrs()' do
     subject(:loa1_user) { build(:user, :loa1) }
     subject(:loa3_user) { build(:user, :loa3) }
+
+    # Possibly consider removing this spec, and changing from_merged_attrs:
+    # def self.from_merged_attrs(existing_user, new_user)
+    #   existing_user.destroy
+    #   new_user.save
+    #   new_user
+    # end
     it 'should not down-level' do
       user = described_class.from_merged_attrs(loa3_user, loa1_user)
       expect(user.loa[:current]).to eq(loa3_user.loa[:current])
       expect(user.loa[:highest]).to eq(loa3_user.loa[:highest])
       expect(user).to be_valid
     end
+
     it 'should up-level' do
       user = described_class.from_merged_attrs(loa1_user, loa3_user)
       expect(user.loa[:current]).to eq(loa3_user.loa[:current])
       expect(user.loa[:highest]).to eq(loa3_user.loa[:highest])
       expect(user).to be_valid
     end
-    it 'should use newer attrs unless they are empty or nil' do
+
+    it 'should use newer attrs even if they are empty or nil' do
       new_user = build(:user, :loa1, first_name: 'George', last_name: 'Washington', gender: '', birth_date: nil)
       user = described_class.from_merged_attrs(loa3_user, new_user)
       expect(user.first_name).to eq('George')
       expect(user.last_name).to eq('Washington')
-      expect(user.gender).to eq(loa3_user.gender)
-      expect(user.birth_date).to eq(loa3_user.birth_date)
-      expect(user.zip).to eq(loa3_user.zip)
+      expect(user.gender).to eq('')
+      expect(user.birth_date).to eq(nil)
+      expect(user.zip).to eq(new_user.zip)
```

Comment:
How do you want to handle the case where the existing user has a value for `loa.highest` but the new saml user comes in with a value of `nil`?  E.g
```ruby
#line 33
new_user = build(:user, :loa1, first_name: 'George', last_name: 'Washington', gender: '', birth_date: nil, loa: {current: 1, highest: nil})
# ... then ...
expect(user.loa[:highest]).to eq(nil)
```

Reply:
So according to Tanel this will never happen. :-p

<img width="754" alt="screen shot 2017-11-14 at 2 26 53 pm" src="https://user-images.githubusercontent.com/591737/32808458-ecc2cad6-c947-11e7-9b0c-dd02d91ffad7.png">

Reply:
I think it doesn't hurt to do null handling in the saml deserializer.

if authn_context = LOA1, and null, then 1
if authn_context = LOA3, and null, then raise an error